### PR TITLE
Addressing Ownership Conflict when creating catalog/schemas

### LINKED ISSUE #2932: [BUG]: `create-catalog-schemas` might fail when transferring ownership of the catalog
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

`create-catalog-schemas` might fail when transferring ownership of the catalog if the transferal of the ownership removes the privileges for the user running the command to apply pending grants.

For example,
- The table mapping contains entries mapping: `hive_metastore.database.table` to `catalog.database.table`
- User B has legacy `OWN` permissions on `hive_metastore.database` 
- User C has legacy `USAGE` permissions on `hive_metastore.database` 
- Above permissions are ordered like this list in the UCX inventory grants table.
- User A runs the `create-catalog-schemas` command, in UC
   1. The catalog `catalog` is created with ownership for user A
   1. The database `database` is created with ownership for user A
   1. The ownership of database `database` is transferred from user A to user B
   1. User C gets `USAGE` permission database `database` (executed by user A) - this might fail(?) as user A is still owner of the catalog but not of the database
   1. The ownership of catalog `catalog` is transferred from user A to user B
   2. User C **fails to** get `USAGE` permissions on catalog `catalog` as user A does not have the permissions to grant these privileges to user C (anymore).

Notes: 
- If the ownership is transferred (correctly) it might still become a problem when migrating tables as the permissions to create UC tables might be missing.
- If the ownership is **not** transferred this might become a data security breach as the user performing the migration receives ownership of data objects it maybe did not had access to before.

Linked issue:
#2890

### Expected Behavior

- The `create-catalog-schemas` command should not fail
- The end state should reflect the right ownership, while not blocking table migration

### Steps To Reproduce

See above

### Cloud

Azure

### Operating System

macOS

### Version

latest via Databricks CLI

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by FastLee**
Closes #2932

### CODE REVIEW

File: src/databricks/labs/ucx/hive_metastore/catalog_schema.py [hunk 1]
```diff
@@ -72,6 +75,16 @@ def _apply_from_legacy_table_acls(self) -> None:
                 continue
             logger.debug(f"Migrating acls on {grant.this_type_and_key()} using SQL query: {acl_migrate_sql}")
             self._backend.execute(acl_migrate_sql)
+        own_grants = [grant for grant in grants if grant.action_type == "OWN"]
+        for grant in own_grants:
```
Comment (nfx) [inline src/databricks/labs/ucx/hive_metastore/catalog_schema.py:79]:
```suggestion
        for grant in grants:
            if grant.action_type != "OWN":
                continue
```
can be simplified

File: src/databricks/labs/ucx/hive_metastore/catalog_schema.py [hunk 2]
```diff
@@ -72,6 +75,16 @@ def _apply_from_legacy_table_acls(self) -> None:
...
+            own_acl_migrate_sql = grant.uc_grant_sql()
+            logger.debug(f"Migrating ownership on {grant.this_type_and_key()} using SQL query: {own_acl_migrate_sql}")
```
Comment (nfx) [inline src/databricks/labs/ucx/hive_metastore/catalog_schema.py:81]: this debug statement does not make sense logically here, move it after the if-continue

File: tests/integration/conftest.py [hunk 3]
```diff
@@ -1244,6 +1244,28 @@ def prepared_principal_acl(runtime_ctx, env_or_skip, make_mounted_location, make
     )
 
 
+@pytest.fixture
+def prepared_legacy_acl(runtime_ctx, make_mounted_location, make_catalog, make_schema):
```
Comment (nfx) [inline tests/integration/conftest.py:1248]: why do you need to introduce new fixture if it's used only once? it hinders readability

Comment (FastLee) [reply tests/integration/conftest.py:1248]: I'm anticipating more tests with legacy ACLs. But will remove for now.

File: src/databricks/labs/ucx/hive_metastore/catalog_schema.py (hunk 2 above)
Comment (FastLee) [reply src/databricks/labs/ucx/hive_metastore/catalog_schema.py:81]: Of course, the if continue was added in the review.

File: src/databricks/labs/ucx/hive_metastore/catalog_schema.py (hunk 1 above)
Comment (FastLee) [reply src/databricks/labs/ucx/hive_metastore/catalog_schema.py:79]: Done.

**Review by nfx — APPROVED** [review]
lgtm

**Pull request merged**